cpp/src/join/hash_join.cu

```diff
@@ -47,34 +47,37 @@ namespace {
  * @brief Calculates the exact size of the join output produced when
  * joining two tables together.
  *
- * @throw cudf::logic_error if JoinKind is not INNER_JOIN or LEFT_JOIN
- *
- * @tparam JoinKind The type of join to be performed
+ * @throw cudf::logic_error if join is not INNER_JOIN or LEFT_JOIN
```

Comment:
```suggestion
 * @throw cudf::logic_error if join is not INNER_JOIN or LEFT_JOIN.
```

Reply:
We use dot for complete sentences which are usually in `@brief` but not in `@param`, `@tparam`, or `@throw`.

Reply:
Yes. But here all the existing tags have period at the end so I want to enforce consistency between the newly added `@param` with the existing one. We should either remove all or keep all.

Reply:
In that sense, I'm inclined to remove the period since they are not complete sentences.